### PR TITLE
Update parent context safety

### PR DESCRIPTION
**Pull request by boyan-velinov**
Jira: LMCROSSITXSADEPLOY-522

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java [hunk 1]
```diff
@@ -64,7 +76,21 @@ private ErrorType getSubProcessErrorType(HistoricProcessInstance subProcess) {
     protected abstract ExecutionStatus onAbort(DelegateExecution context, ErrorType errorType) throws MonitoringException;
 
     protected ExecutionStatus onSuccess(DelegateExecution context) {
+        duplicateProcessVariablesFromSubProcess(context);
         return ExecutionStatus.SUCCESS;
     }
 
+    private void duplicateProcessVariablesFromSubProcess(DelegateExecution context) {
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java]: The name of the method is a little bit tricky. Could we rename it?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java [hunk 2]
```diff
@@ -64,7 +76,21 @@ private ErrorType getSubProcessErrorType(HistoricProcessInstance subProcess) {
...
+        List<String> processVariablesToDuplicate = getProcessVariablesToDuplicate();
+        for (String processVariable : processVariablesToDuplicate) {
+            HistoricVariableInstance historicVariableInstance = activitiFacade.getHistoricVariableInstance(subProcessId, processVariable);
+            if (historicVariableInstance != null) {
+                context.setVariable(processVariable, historicVariableInstance.getValue());
+            }
+        }
+    }
+
+    protected abstract List<String> getProcessVariablesToDuplicate();
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java]: The same here... processVariablesToDuplicate is not talking much in the Activiti context

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java (hunk 1 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java (hunk 2 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java]: Done

**Pull request merged**